Source/Engine/Utilities/Extensions.cs

```diff
@@ -404,5 +238,184 @@ public static void Shuffle<T>(this Random random, IList<T> collection)
+                output.Z = NextFloat(random) * 2.0f - 1.0f;
+
+            } while (output.LengthSquared > 1 || output.LengthSquared < 1e-6f);
+
+            output *= (1.0f / (float)Math.Sqrt(output.LengthSquared));
+
+            return output;
+        }
+
+        /// <summary>
+        /// Generates a uniformly distributed random unit length vector point on a unit sphere in 2D.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <returns>A random <see cref="Vector2"/>.</returns>
+        public static Vector2 NextVector2(this Random random)
+        {
+            Vector2 output;
+
+            do
+            {
+                output.X = NextFloat(random) * 2.0f - 1.0f;
+                output.Y = NextFloat(random) * 2.0f - 1.0f;
+
+            } while (output.LengthSquared > 1 || output.LengthSquared < 1e-6f);
+
+            output *= (1.0f / (float)Math.Sqrt(output.LengthSquared));
+
+            return output;
+        }
+
+        /// <summary>
+        /// Generates a random <see cref="Color"/>.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="trueRandom">Should the color be generated from a single random Hue value or separate values for each channel.</param>
+        /// <param name="randomAlpha">Randomize the alpha value.</param>
+        /// <returns>A nice random <see cref="Color"/>.</returns>
+        public static Color NextColor(this Random random, bool trueRandom, bool randomAlpha)
+        {
+            float alpha = randomAlpha ? NextFloat(random) : 1f;
+
+            if (!trueRandom)
+                return Color.FromHSV(NextFloat(random, 0f, 360f), 1f, 1f, alpha);
+
+            return new Color(
+                NextFloat(random, 0f, 255f),
+                NextFloat(random, 0f, 255f),
+                NextFloat(random, 0f, 255f), alpha);
+        }
+
+        /// <summary>
+        /// Gets a random <see cref="double"/>.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="min">The minimum value.</param>
+        /// <param name="max">The maximum value.</param>
+        /// <returns>A random <see cref="double"/>.</returns>
+        public static double NextDouble(this Random random, double min = 0.0d, double max = 2.0d)
+        {
+            return random.NextDouble() * (max - min) + min;
+        }
+
+        /// <summary>
+        /// Gets a random <see cref="long"/>.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <returns>A random <see cref="long"/>.</returns>
+        internal static long NextLong(this Random random)
+        {
+            var numArray = new byte[8];
+            random.NextBytes(numArray);
+            return (long)(BitConverter.ToUInt64(numArray, 0) & 9223372036854775807L);
```

Comment:
For the number you should probably use long.MaxValue to make it more clear.

Reply:
Done.